ksqldb-engine/src/main/java/io/confluent/ksql/engine/SandboxedExecutionContext.java

```diff
@@ -162,19 +162,23 @@ public ExecuteResult execute(
       final ServiceContext serviceContext,
       final ConfiguredKsqlPlan ksqlPlan
   ) {
-    final ExecuteResult result = EngineExecutor.create(
-        engineContext,
-        serviceContext,
-        ksqlPlan.getConfig()
-    ).execute(ksqlPlan.getPlan());
+    try {
+      final ExecuteResult result = EngineExecutor.create(
+          engineContext,
+          serviceContext,
+          ksqlPlan.getConfig()
+      ).execute(ksqlPlan.getPlan());
 
-    // Having a streams running in a sandboxed environment is not necessary
-    if (!getKsqlConfig().getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)) {
-      result.getQuery().map(QueryMetadata::getKafkaStreams).ifPresent(streams -> streams.close());
-    } else {
-      engineContext.getQueryRegistry().closeRuntimes();
+      // Having a streams running in a sandboxed environment is not necessary
+      if (!getKsqlConfig().getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)) {
+        result.getQuery().map(QueryMetadata::getKafkaStreams).ifPresent(streams -> streams.close());
+      }
+      return result;
+    } finally {
+      if (getKsqlConfig().getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)) {
+        engineContext.getQueryRegistry().closeRuntimes();
```

Comment:
At this point we really can close all the runtimes, right? We don't need them for anything? And is there any risk that we'll miss the `ifPresent(streams -> streams.close())` above and would want to do that cleaning here too?

Reply:
This is validation so yes we won't use them anymore. And the old engine didn't really care but now we need the cleaned up